### ISSUE TITLE
Can't move certain windows unless modifier key is used

### ISSUE THREAD

**Issue by mikedebian** [window-maker/wmaker#20]
Example case: Telegram desktop has its own window decoration, which renders right clicking and selecting window attributes unobtainable. Window can not be manipulated in the common way (resize, moving), although window title bar buttons work. Have to use the ALT key to move the window about.

Only way to do it is to right click on the icon and select attributes and deselect the options (hide title bar, etc).

Even then, the buttons will not show correctly, however the window will have it's own (double) titlebar.

Perhaps expected default should be not to disable the window titlebar?

**Comment by dmaciejak** [comment]
Telegram desktop is created without titlebar, border, standard close button, etc.
If you want to override that you can iconify the window and change the "Attributes", from "Advanced Options" menu entry, check the "Ignore decoration changes" Apply and Save.
Then dock it, next time you will open the Telegram desktop you will have the window titlebar and will be able to drag the window as usual.

**Comment by gryf** [comment]
You can also use the shortcut key to access window menu (by default it is `ctrl-esc`), although iconify and then using right mouse click on it will also do.

FTR - there are more and more applications which try to force their own way for managing their windows, GTK3/4 apps are notable examples. Usually it's enough to change the settings on the Attributes->Window Settings, but some of them needs the "Ignore decoration changes" mentioned by David, to force them to behave.

Besides that we could apply workarounds to enable window decorations (like tiitle/resize bars) it's a bug, as for whatever reason wmaker doesn't allow to pass the control to the apps without decorations.

**Comment by gryf** [comment]
I know what root cause is for self-managed applications, i.e. those which have their own titlebars-like elements, resize grips and so on. I started to work on the solution.

**Comment by cpchan** [comment]
Go to WPref -> "Expert User Preferences" (the one with the scholastic hat) -> check "Ignore decoration hints for GTK applications"

Does that help you?

**Comment by mikedebian** [comment]
Yes, issue seems to be solved.

**Comment by gryf** [comment]
No, it's not solved. Window Maker doesn't support `_NET_MOVERESIZE_WINDOW` right now, and this is the reason, why certain windows, which suppose to be controllable by their own decorations are not behave correctly.

I've started investigation and implementation for support this kind of window messages, although I've stuck with it as seems like the X events are not exposed to window manager when mouse cursor is within the window.

**Comment by dmaciejak** [comment]
Should be fixed, see https://repo.or.cz/wmaker-crm.git/commit/6d0953bc229c8b6e8ed0cd251bd80280acf3e7e8
